packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -140,7 +173,10 @@ const AccountSignup = ({ enableApp, isModalVisible, clients_country, onSignup, r
                                         })}
                                         type='button'
                                         is_disabled={!values.residence || !!errors.residence}
-                                        onClick={onResidenceSelection}
+                                        onClick={() => {
+                                            onResidenceSelection();
+                                            setSelectedResidence(values.residence);
+                                        }}
```

Comment:
```suggestion
                                        onClick={() => onResidenceSelection(values)}
```
Now ` setSelectedResidence(values.residence);` could be inside `onResidenceSelection` function.

Reply:
fixed